### ISSUE TITLE
Unable to select item in Autcomplete with keyboard

### ISSUE THREAD

**Issue by jockis** [primefaces/primereact#6308]
### Describe the bug

It's not possible to select an item in Autocomplete that is not an Option group because index get incorrect value in createListItem

### PrimeReact version

10.6.2

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

Go to https://primereact.org/autocomplete/#objects select input, write "sa", use arrows to select an item and press enter
item is not visible in input

### Expected behavior

item should be visible in input
